pkg/controller/node/node_controller.go

```diff
@@ -1270,6 +1270,10 @@ func getAllCandidateMachines(layered bool, config *mcfgv1.MachineOSConfig, build
 		return nil, 0
 	}
 	capacity -= failingThisConfig
+
+	if capacity < 0 {
```

Comment:
I think this should be safe, but just curious if this was added due the kube bump

Reply:
I added it get rid of the gosec warning